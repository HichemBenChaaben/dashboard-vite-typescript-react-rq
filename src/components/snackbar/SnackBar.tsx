import React, { ReactNode, useState } from "react";

interface SnackBarContentProps {
  children: ReactNode;
}

interface SnackBarIndicatorProps {
  children: ReactNode;
  [key: string]: any;
}

const SnackBar: React.FC<SnackBarIndicatorProps> & {
  Content: React.FC<SnackBarContentProps>;
} = ({ children, ...props }) => {
  const [visible, toggleVisible] = useState(true);

  if (!visible) return null;

  return (
    <div
      data-testid="snack-bar"
      className="fixed bottom-4 right-4 w-[400px] p-4 bg-gray-800 text-white rounded-lg shadow-lg flex items-start space-x-4 md:flex"
      {...props}
    >
      {children}
      <button
        data-testid="snack-bar-close-btn"
        onClick={() => toggleVisible(false)}
        className="p-2 rounded-sm hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-gray-500"
      >
        <span className="text-white text-xl fa fa-times" />
      </button>
    </div>
  );
};

const SnackBarContent: React.FC<SnackBarContentProps> = ({ children }) => (
  <div className="flex-1 text-sm font-medium">{children}</div>
);

SnackBar.Content = SnackBarContent;

export default SnackBar;
