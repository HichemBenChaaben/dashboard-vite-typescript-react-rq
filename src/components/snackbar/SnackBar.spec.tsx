import "@testing-library/jest-dom/vitest";
import { render, screen, waitFor } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import SnackBar from "@/components/snackbar/SnackBar";

describe("SnackBar component", () => {
  it("should render correctly", () => {
    render(<SnackBar>Hotkeys content</SnackBar>);
    expect(screen.getByRole("button")).toBeInTheDocument();
    expect(screen.getByText("Hotkeys content")).toBeInTheDocument();
  });
  it("should match the snapthot", () => {
    expect(render(<SnackBar>test</SnackBar>)).toMatchSnapshot();
  });
  it("should hide when pressing the close button", async () => {
    render(<SnackBar>test</SnackBar>);
    const closeButton = screen.getByTestId("snack-bar-close-btn");
    await closeButton.click();
    waitFor(() => {
      expect(screen.queryByTestId("snack-bar")).toBeNull();
    });
  });
});
