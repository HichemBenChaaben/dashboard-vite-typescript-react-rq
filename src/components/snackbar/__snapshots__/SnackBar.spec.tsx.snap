// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SnackBar component > should match the snapthot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fixed bottom-4 right-4 w-[400px] p-4 bg-gray-800 text-white rounded-lg shadow-lg flex items-start space-x-4 md:flex"
        data-testid="snack-bar"
      >
        test
        <button
          class="p-2 rounded-sm hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-gray-500"
          data-testid="snack-bar-close-btn"
        >
          <span
            class="text-white text-xl fa fa-times"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fixed bottom-4 right-4 w-[400px] p-4 bg-gray-800 text-white rounded-lg shadow-lg flex items-start space-x-4 md:flex"
      data-testid="snack-bar"
    >
      test
      <button
        class="p-2 rounded-sm hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-gray-500"
        data-testid="snack-bar-close-btn"
      >
        <span
          class="text-white text-xl fa fa-times"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
